udata/core/dataset/models.py

```diff
@@ -217,6 +217,27 @@ def check_availability(self):
         '''
         return self.extras.get('check:available', 'unknown')
 
+    def is_need_check(self):
+        '''Does the resource needs to be checked against its linkchecker?
+
+        We check unavailable resources often. Available resources are checked
+        less and less frequently based on their historical availability.
+        '''
+        cache_duration = current_app.config['LINKCHECKING_MIN_CACHE_DURATION']
+        count_availability = self.extras.get('check:count-availability', 1)
+        is_available = self.check_availability()
+        if is_available == 'unknown':
+            return True
+        if is_available:
```

Comment:
`elif` (I know it produces the same result but it's just to mark the 3 states)

Reply:
👍